src/stan/services/util/generate_transitions.hpp

```diff
@@ -44,14 +44,17 @@ void generate_transitions(stan::mcmc::base_mcmc& sampler, int num_iterations,
                           util::mcmc_writer& mcmc_writer,
                           stan::mcmc::sample& init_s, Model& model,
                           RNG& base_rng, callbacks::interrupt& callback,
-                          callbacks::logger& logger) {
+                          callbacks::logger& logger, size_t n_chain = 0) {
   for (int m = 0; m < num_iterations; ++m) {
     callback();
 
     if (refresh > 0
         && (start + m + 1 == finish || m == 0 || (m + 1) % refresh == 0)) {
       int it_print_width = std::ceil(std::log10(static_cast<double>(finish)));
       std::stringstream message;
+      if (n_chain > 0) {
+        message << "Chain [" << (n_chain + 1) << "]";
```

Comment:
This makes sense cause I think cmdstanr/cmdstanpy append similar things, but let's check with the format.